src/main/java/com/cloudbees/jenkins/plugins/advisor/Reminder.java

```diff
@@ -15,65 +15,35 @@
  * Displays the reminder that the user needs to register.
  */
 @Extension
-public class Reminder extends PageDecorator {
+public class Reminder extends AdministrativeMonitor {
 
-  private transient volatile long lastNagTime;
-
-  public Reminder() {
-    load();
-  }
-
-  public boolean isNagDue() {
+  @Override
+  public boolean isActivated() {
     if (!(Jenkins.getInstance().servletContext.getAttribute("app") instanceof Jenkins)) {
       return false;   // no point in nagging the user during licensing screens
     }
 
     AdvisorGlobalConfiguration config = AdvisorGlobalConfiguration.getInstance();
     if (config.isValid()) {
       return false; // no nag when registered
-    }
-    if (!config.isPluginEnabled()) {
+    } else if (!config.isPluginEnabled()) {
       return false; // no nag when disabled
-    }
-    if (config.isNagDisabled()) {
+    } else if (config.isNagDisabled()) {
       return false; // no nag when explicitly avoided
+    } else {
+      return true;
     }
-    try {
-      HttpSession session = Stapler.getCurrentRequest().getSession(false);
-      if (session != null) {
-        Long nextNagTime = (Long) session.getAttribute(Reminder.class.getName() + ".nextNagTime");
-        if (nextNagTime != null) {
-          return System.currentTimeMillis() > nextNagTime;
-        }
-      }
-      return System.currentTimeMillis() >= lastNagTime + TimeUnit.SECONDS.toMillis(600);
-    } catch (Exception ex) {
-      // If there's an issue trying to figure out if we need to nag, let's nag
-    }
-    return true;
-  }
-
-  public static Reminder getInstance() {
-    for (PageDecorator d : PageDecorator.all()) {
-      if (d instanceof Reminder) {
-        return (Reminder) d;
-      }
-    }
-    throw new AssertionError(Reminder.class + " is missing its descriptor");
   }
 
-  public HttpResponse doAct(StaplerRequest request, @QueryParameter(fixEmpty = true) String yes,
+  protected HttpResponse doAct(StaplerRequest request, @QueryParameter(fixEmpty = true) String yes,
```

Comment:
@kwhetstone can we add `@RequirePOST` to satisfy the CSRF concerns? cc @daniel-beck